clients/src/test/java/org/apache/kafka/clients/consumer/ConsumerRecordsTest.java

```diff
@@ -145,6 +144,26 @@ public void testRecordsByTopic() {
         }
     }
 
+    @Test
+    public void testRecordsAreImmutable() {
+        String topic = "topic";
+        int recordSize = 3;
+        int partitionSize = 6;
+        int emptyPartitionIndex = 2;
+
+        ConsumerRecords<Integer, String> records = buildTopicTestRecords(recordSize, partitionSize, emptyPartitionIndex, Collections.singleton(topic));
+        ConsumerRecords<Integer, String> emptyRecords = new ConsumerRecords<>(Collections.emptyMap());
+
+        assertThrows(UnsupportedOperationException.class, () -> records.records(new TopicPartition(topic, 0))
+            .add(new ConsumerRecord<>(topic, 0, 0, 0L, TimestampType.CREATE_TIME,
```

Comment:
nit:
`new ConsumerRecord(...)` has the same situation as `new TopicPartition(topic, 0)`.